### ISSUE TITLE
auto return type fn definition not allowed -- unless static method in class

### ISSUE THREAD

**Issue by ddheart** [jupyter-xeus/xeus-cling#321]
Hello, after installing xeus-cling from condaforge (xeus-cling version 0.8.1, cling 0.6), I've observed this behavior: 

inside a cell --
```
class S {
public:
  static auto fn() -> double {
      return 3.2;
  }  
};
cout << S::fn() << endl;
```
Output: 3.2

but by itself, 
```
auto fn() -> double {
   return 3.2;
}
```
yields error "function definition is not allowed here"?

Unsure if this is an error w xeus-cling or cling, and didn't find clarifying documentation so posting here. Apologies if this is old news. Not sure when this is expected behavior versus not, but this specific type of declaration is acting like it's encircled by main().

**Comment by SylvainCorlay** [comment]
Indeed, this is a parsing issue in cling. Often, grouping with other statements will fix this issue.

**Comment by ddheart** [comment]
Ah I see. Tried grouping it other statements in the same cell but error persists. Could you elaborate or give a small example? and/or a link to a cling thread/issue about this variant of cling parsing issues?
